### PR TITLE
convert a few various to callnatives

### PR DESCRIPTION
**Pull request by ghoulslash**
Convert the following various to callnatives

- VARIOUS_SET_Z_EFFECT
- VARIOUS_TRY_SYMBIOSIS
- VARIOUS_CAN_TELEPORT
- VARIOUS_GET_BATTLER_SIDE
- VARIOUS_CHECK_PARENTAL_BOND_COUNTER

### CODE REVIEW

**Comment by Jaizu** [comment]
Please fix conflicts.

**Comment by AsparagusEduardo** [comment]
Please solve conflicts.

**Comment by ghoulslash** [comment]
> Please solve conflicts.

Done

**Comment by equezada-jej** [comment]
I know this was introduced later, but we should use `NATIVE_ARGS` ^^;

**Comment by LOuroboros** [comment]
Also, instead of increasing `gBattlescriptCurrInstr` by arbitrary amounts, wouldn't it be better to do `gBattlescriptCurrInstr = cmd->nextInstr;`? :eyes:

**Comment by equezada-jej** [comment]
> Also, instead of increasing `gBattlescriptCurrInstr` by arbitrary amounts, wouldn't it be better to do `gBattlescriptCurrInstr = cmd->nextInstr;`? 👀
> 
> **EDIT**: I know they're not arbitrary per se, but still. Is it not possible to do that in these cases?

This is part of what I mean by using `NATIVE_ARGS` ^^;

**Comment by AsparagusEduardo** [comment]
> Also, instead of increasing `gBattlescriptCurrInstr` by arbitrary amounts, wouldn't it be better to do `gBattlescriptCurrInstr = cmd->nextInstr;`? 👀
> 
> **EDIT**: I know they're not arbitrary per se, but still. Is it not possible to do that in these cases?

This is part of what I mean by using `NATIVE_ARGS` ^^;

**Comment by ghoulslash** [comment]
> I know this was introduced later, but we should use `NATIVE_ARGS` ^^;

good call. just updated with them. The SetZEffect one still uses the old method but is a special case. would probably need a separate PR to clean it up if we wanted to down the line.

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -8805,7 +8805,7 @@ static bool32 CanTeleport(u8 battlerId)
 {
     struct Pokemon *party = GetBattlerParty(battlerId);
     u32 species, count, i;
-
+
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:8808]: please remove trailing whitespace.

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -16317,33 +16317,33 @@ static bool8 IsFinalStrikeEffect(u16 move)
 // 10 bytes long (callnative(5) + counter(1) + ptr(4))
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:16317]: Comment no longer needed.

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -16317,33 +16317,33 @@ static bool8 IsFinalStrikeEffect(u16 move)
...
-    if (gSpecialStatuses[gBattlerAttacker].parentalBondState == counter && gBattleMons[gBattlerTarget].hp != 0)
-        gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 6);
+    if (gSpecialStatuses[gBattlerAttacker].parentalBondState == cmd->counter && gBattleMons[gBattlerTarget].hp != 0)
+        gBattlescriptCurrInstr = cmd->jumpInstr;
     else
-        gBattlescriptCurrInstr += 10;
+        gBattlescriptCurrInstr = cmd->nextInstr;
 }
 
 // 6 bytes long (callnative(5) + battler(1))
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:16329]: Comment no longer needed.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by ghoulslash** [comment]
ready for review

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**